### PR TITLE
Add a Local Test binary, to enable local TerminalApp testing

### LINKED ISSUE #1210: Re-enable TerminalApp tests `SettingsTests` and `TabsTests`
This is a follow up from #1164. 

Right now our CI isn't running a windows version high enough to be able to run these tests, so they're commented out. 

Once #1012 is checked in, we can re-enable those tests.

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

In #1164 we learned that our CI doesn't support WinRT testing. This made us all sad. Since that merged, we haven't really added any TerminalApp tests, because it's a little too hard. You'd have to uncomment the entire file, and if the list of types changed you'd have to manually update the sxs manifest and appxmanifest.

Since that was all insane, I created a new Terminal App unittesting project without those problems.
1. The project is not named *Unit*Test*, so the CI won't run it, but it will run locally.
2. The project will auto-generate its SxS manifest, using the work from #1987. 
3. We'll use the SxS manifest from step 2 to generate an AppxManifest for running packaged tests.

## PR Checklist
* [x] Closes #1210 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] **Tests added/passed**
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

I'm 90% just opening this PR to make sure it works in CI.

### CODE REVIEW

File: src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  <!-- Step 1: Combine all our SxS manifests into a single SxS manifest. TAEF
+  needs us to specify a single activation context at runtime, so we need a
+  single file with all our types in it.-->
+  <Target Name="_LocalTestsGenerateCombinedManifests"
+          Inputs="@(_ConsoleWinmdManifest)"
+          Outputs="$(OutDir)$(TargetName).manifest"
+          DependsOnTargets="_ConsoleGenerateAdditionalWinmdManifests">
+
+    <Exec Command="mt.exe -manifest @(_ConsoleWinmdManifest, ' -manifest ') -out:$(OutDir)$(TargetName).manifest" />
```
Comment (DHowett-MSFT) [inline src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj]: i am so happy to see you writing custom msbuild targets
it is like an infection

File: src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  </Target>
+
+  <!-- Manually copy all our dependent DLLs into our OutDir. For SxS activation
+  to work, they all need to be in the same directory as our test dll. -->
+  <ItemDefinitionGroup>
+    <PostBuildEvent>
+      <Command>
+        echo OutDir=$(OutDir)
+
+        (xcopy /Y $(_CppWinrtBinRoot)TerminalConnection\TerminalConnection.dll &quot;$(OutDir)\TerminalConnection.dll*&quot; )
```
Comment (DHowett-MSFT) [inline src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj]: these should be a target containing `<Copy`.

File: tools/GenerateAppxFromManifest.ps1 [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# This script is used for taking all the activatable classes from a SxS manifest
+# and adding them as Extensions to an Appxmanifest.xml.
+# Params:
+#  - SxSManifest: The path to the SxS manifest to get the types from
+#  - AppxManifestPrototype: The path to an AppxManifest.xml-style XML document to add the Extensions to
+#  - SxSManifest: The path to write the updated XML doc to.
+
+param (
+    [parameter(Mandatory=$true, Position=0)]
```
Comment (DHowett-MSFT) [inline tools/GenerateAppxFromManifest.ps1]: When calling this script, please use named parameters: `-SxSManifest whatever.xml -AppxManifestPrototype whatever2.xml`

File: tools/GenerateAppxFromManifest.ps1 [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+}
+
+# Add our fully constructed list of extensions to the original Appxmanifest prototype
+$appxPrototypeData.Package.AppendChild($Extensions) | Out-Null
+
+# Write the modified xml back out.
+$appxPrototypeData.save($OutPath)
+
+# Left as a helper for debugging:
```
Comment (DHowett-MSFT) [inline tools/GenerateAppxFromManifest.ps1]: technically you can make it do this if you pass `-debug` but that's extra credit. loko up cmdlet binding verbose debug stuff.

File: src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/LocalTests_TerminalApp/TerminalApp.LocalTests.vcxproj]: I'm not pleased that this is something we have to do at all, but I concur it's a lot smoother than xcopying files all over

**Pull request merged**